src/angular-local-storage.js

```diff
@@ -318,9 +320,23 @@ angular
             }
             if (!!key) {
               var cookiePath = "; path=" + cookie.path;
-              if(cookie.domain){
+              if (cookie.domain) {
                 cookieDomain = "; domain=" + cookie.domain;
               }
+              /* Providing the secure parameter always takes precedence over config 
+               * (allows developer to mix and match secure + non-secure) */
+              if (typeof secure === 'boolean') {
+                  if (secure === true) {
+                      /* We've explicitly specified secure,
+                       * add the secure attribute to the cookie (after domain) */
+                      cookieDomain += "; secure";
+                  }
+                  // else - secure has been supplied but isn't true - so don't set secure flag, regardless of what config says
+              }
+              else if (typeof secure === 'boolean' && cookie.secure === true) {
```

Comment:
The `typeof secure === 'boolean'` should not be needed here. We reach this `else if` if `secure` is not a `boolean`. Besides this your explanation makes total sense. Thanks for the input :+1: